### DISCUSSION TITLE
有没有react-resizeable的替代品？

### DISCUSSION THREAD

**Discussion by vaynevayne** [ant-design/ant-design#42446] [Q&A]
这个库需要显示width值。而且会实时修改width，导致状态一直更新 都没办法做优化，问一下有没有其他的实现呢

**Comment by yoyo837** [comment]
我就用的 react-resizable 没遇到你说的问题

**Reply by vaynevayne** [reply]
onResize的时候 你console试试 这里面得setState

**Reply by yoyo837** [reply]
So what?

**Comment by q269384828** [comment]
你说的是 table 的表头拖拽调整大小是吧,  这个用 react-resizable 确实会有很多问题, 表格内部复杂一点点都麻烦,  我记得之前官方是有打算做的.

**Reply by vaynevayne** [reply]
官方没做, 话说一个功能完整的table, 功能也就那么多, 但是目前都没有什么好的组件支持

**Comment by linxianxi** [comment]
可以不实时改变宽度，拖动的时候，显示一个竖线，放手的时候再更新宽度

**Reply by linxianxi** [reply]
类似这种  https://table.rsuitejs.com/#resizable

**Reply by yoyo837** [reply]
改官网的 react-resizeable demo是可以实现的,只是没有线, 线要额外搞

**Reply by vaynevayne** [reply]
https://daybrush.com/moveable/storybook/index.html?path=/story/basic--basic-resizable 这个库会不会比 react-resizeable 更好, 但是不知道怎么对接

不过对于实时宽度的问题, 可以单独声明一个 state 来同步
https://github.com/vaynevayne/soul/blob/main/packages/soul-core/src/Table/Table.tsx#L153

**Comment by q269384828** [comment]
` react-resizable `最大的问题倒不是说频繁更新width导致的性能问题, 而是  rc-table 内部对列宽本身有一套计算的逻辑和更新的规则,  这导致 table 复杂起来之后拖动可能存在漂移的问题.

**Comment by q269384828** [comment]
> 类似这种 https://table.rsuitejs.com/#resizable

这个方案在宽度不满的时候就直接摆烂了.

**Reply by linxianxi** [reply]
什么意思
